src/lib/eslint/ESLintEngine.ts

```diff
@@ -21,74 +109,137 @@ export class ESLintEngine implements RuleEngine {
 		if (this.initialized) {
 			return;
 		}
-
 		this.logger = await Logger.child(this.getName());
 
+		this.fileHandler = new FileHandler();
+		this.config = await Controller.getConfig();
+
+		// We currently assume you must execute the engine from your project directory.  This doesn't matter for all
+		// situations, but it is critical at least for typescript.  TS rules require tsconfig.  In the future we could
+		// change to treat targetPaths which are directories as project directories, and look inside them for a
+		// tsconfig.json file.  Until then, just require that if you want to run against *.ts, you better do so from a
+		// working directory that contains tsconfig.
+		this.esconfig = await this.fileHandler.exists("tsconfig.json") ?
```

Comment:
I'm confused - if `tsconfig.json` file exists, why aren't we using it?

Reply:
We are using it.  If it exists, we use the default eslint config DEFAULT_ESCONFIG_TS which references tsconfig.json.  If not, we use the simpler DEFAULT_ESCONFIG which does not.